File: Construct-K-Palindrome-1400.py
from collections import Counter

class Solution(object):
    def canConstruct(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: bool
        """

        l = len(s)

        if l == k:
            return True

        if l < k:
            return False

        freq = Counter(s)

        odd = sum(1 for count in freq.values() if count % 2 == 1)

        return odd <= k

File: Construct-K-Palindrome-1400.java
import java.util.HashMap;

class Solution {
    public boolean canConstruct(String s, int k) {

        int len = s.length();
        if(len == k) return true;
        if(len < k) return false;

        HashMap<Character , Integer> map = new HashMap<>();
        
        for(char c : s.toCharArray()){
            map.put(c , map.getOrDefault(c , 0) + 1);
        }

        int odd = 0 ;

        for(int val: map.values()){
            if(val % 2 != 0){
                odd++;
            }
        }

        if(odd > k){
            return false;
        }

        return true; 
    }
}